mcap_vendor/CMakeLists.txt

```diff
@@ -18,7 +20,7 @@ if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_EXTENSIONS OFF)
 endif()
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  add_compile_options(-Wall -Wextra -Wpedantic -Werror)
```

Comment:
Why the removal of `-Werror`? Was something failing?

Reply:
As a general policy, I think using `-Werror` is a mistake, and we don't use it anywhere else in the ROS 2 core.

The problem is that while we can, and do, make things compile warning-free on our Tier-1 platforms, ROS 2 is used across a wide variety of platforms, with a wide variety of compilers.  Turning warnings into errors just makes it annoying for those people not using our Tier-1 platforms.